aplib.net-demo/Assets/Scripts/Points/PointsManager.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using UnityEngine;
+
+/// <summary>
+/// Manages the player's points.
+/// </summary>
+public class PointsManager
+{
+    /// <summary>
+    /// Event that is invoked when points are added.
+    /// </summary>
+    /// <param name="points">The current amount of points.</param>
+    public event Action<int> PointsAdded;
+
+    /// <summary>
+    /// The amount of points the player has.
+    /// </summary>
+    [field: SerializeField]
+    public int Points { get; private set; }
+
+    /// <summary>
+    /// The singleton instance of this class.
+    /// </summary>
+    public static PointsManager Instance
+    {
+        get
+        {
+            _instance ??= new PointsManager();
+            return _instance;
+        }
+        private set => _instance = value;
+    }
+
+    private static PointsManager _instance;
+
+    [SerializeField]
+    private readonly int _unusedItemPoints = 5;
+
+    /// <summary>
+    /// Private constructor to prevent instantiation by other classes.
+    /// </summary>
+    private PointsManager() { }
```

Comment:
Do we need to enforce this? I'm not saying we don't, but I'm wondering about the reasoning.

Reply:
Yes, private constructors are part of the standard Singleton pattern iirc